app/components/channel_drawer/channel_list.js

```diff
@@ -287,6 +399,17 @@ class ChannelList extends React.Component {
         return data;
     };
 
+    modalOnChange(option) {
+        this.props.handleDisableDrawer(false);
+        if (option) {
+            option.action();
+        } else {
+            this.setState({
```

Comment:
I think you have every option explicitly closing the modal, so you could change this to
```
this.setState({showOptions: false});

if (option) {
    option.action();
}
```
and get rid of that duplicated code above.